services/graph/pkg/service/v0/sharedwithme.go

```diff
@@ -249,22 +267,27 @@ func (g Graph) listSharedWithMe(ctx context.Context) ([]libregraph.DriveItem, er
 				}
 			}
 
-			driveItems[i] = *driveItem
+			ch <- *driveItem
 
 			return nil
 		})
 	}
 
 	// wait for concurrent requests to finish
-	err = group.Wait()
+	go func() {
+		err = group.Wait()
```

Comment:
i would prefer to wrap the error to not shadow previous appearances  

```go

err = errors.Joins(err, group.Wait())

```

join ignores nil errors

Reply:
Hm, which previous error should that be?

Also I think we need redeclare `err` inside the `group.Go` above. The goroutines shouldn't be sharing the same err variable, I guess.

Reply:
it was just in case if some error happen in the future, ignore it, not crucial